x/dclupgrade/types/message_propose_upgrade_test.go

```diff
@@ -0,0 +1,154 @@
+package types
+
+import (
+	"testing"
+	"time"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+	testconstants "github.com/zigbee-alliance/distributed-compliance-ledger/integration_tests/constants"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/testutil/sample"
+)
+
+func TestMsgProposeUpgrade_ValidateBasic(t *testing.T) {
+	// negative test constants
+
+	planNameLen0 := Plan{
+		Name:   "",
+		Height: 1,
+		Info:   "Some info",
+	}
+
+	planHeight0 := Plan{
+		Name:   "Some plan name",
+		Height: 0,
+		Info:   "Some info",
+	}
+
+	planHeightLess0 := Plan{
+		Name:   "Some plan name",
+		Height: -1,
+		Info:   "Some info",
+	}
+
+	planTimeIsNot0 := Plan{
+		Name:   "Some plan name",
+		Height: 1,
+		Info:   "Some info",
+		Time:   time.Now(),
+	}
+	planUpgradedClientStateIsNotNil := Plan{
+		Name:                "Some plan name",
+		Height:              1,
+		Info:                "Some info",
+		UpgradedClientState: &codectypes.Any{TypeUrl: "333"},
+	}
+
+	// positive test constants
+
+	planNormal := testconstants.Plan
+
+	planInfoLen0 := Plan{
+		Name:   "Some plan name",
+		Height: 1,
+		Info:   "",
+	}
+
+	negative_tests := []struct {
+		name string
+		msg  MsgProposeUpgrade
+		err  error
+	}{
+		{
+			name: "invalid address",
+			msg: MsgProposeUpgrade{
+				Creator: "invalid_address",
+				Plan:    testconstants.Plan,
+			},
+			err: sdkerrors.ErrInvalidAddress,
+		},
+		{
+			name: "omitted address",
```

Comment:
Should we have a test with omitted plan field?

Reply:
No, because `plan` is a struct, not a pointer to a struct.